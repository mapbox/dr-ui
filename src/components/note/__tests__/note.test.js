import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './note-test-cases.js';

describe('note', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.noImage.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.noImage;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.custom.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.custom;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.warning.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.warning;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.legacy.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.legacy;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.error.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.error;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.new.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.new;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.beta.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.beta;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.download.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.download;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
