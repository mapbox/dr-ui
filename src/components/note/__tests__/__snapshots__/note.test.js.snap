// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`note A legacy note, has same styling as warning renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#feefe2",
      "color": "#78471c",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-orange-light round-full color-orange flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Warning
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`note A note to display an error with steps or links on how to troubleshoot. renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#fbe5e5",
      "color": "#9b3134",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-red-light round-full color-red flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Error
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`note A note to display with a message about new products or features. renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#e8f5ee",
      "color": "#15603d",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-green-light round-full color-green flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-plus"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      New!
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`note A note to flag information about a beta release/product renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#f1f3fd",
      "color": "#0c248d",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-blue-light round-full color-blue flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-lightning"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Beta
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`note A warning note to let the user know something has changed or will change. renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#feefe2",
      "color": "#78471c",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-orange-light round-full color-orange flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-alert"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Be careful
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`note Default note renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#f4f7fb",
      "color": "#425870",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-gray-light round-full color-gray flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-book"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Note
    </div>
    <p>
      Here is a little thing to note.
    </p>
  </div>
</div>
`;

exports[`note Note with custom title. renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#f4f7fb",
      "color": "#425870",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-gray-light round-full color-gray flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-book"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      A very important note
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;

exports[`note download renders as expected 1`] = `
<div
  className="dr-ui--note py18 px18 round flex-parent flex-parent--row mb18"
  style={
    Object {
      "background": "#f2effa",
      "color": "#452f92",
    }
  }
>
  <div
    className="flex-child mr18 none block-mm pt3"
  >
    <div
      className="bg-purple-light round-full color-purple flex-parent flex-parent--center-main flex-parent--center-cross"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="events-none icon"
        focusable="false"
        role="presentation"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <use
          xlinkHref="#icon-arrow-down"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </div>
  </div>
  <div
    className="flex-child prose"
  >
    <div
      className="txt-bold txt-m mb6"
    >
      Download from vision.mapbox.com/install
    </div>
    <div>
      <p>
        You can create custom styles with
         
        <a
          href="https://studio.mapbox.com"
        >
          Mapbox Studio
        </a>
         and then add them to your app. To programmatically add one of your custom styles to your
         
        <code>
          mapboxMap
        </code>
        , head to your
         
        <a
          href="https://studio.mapbox.com/styles/"
        >
          styles page
        </a>
        , copy your style's 
        <a
          href="/help/glossary/style-url/"
        >
          style URL
        </a>
        , and then add it to your 
        <code>
          mapboxMap
        </code>
         object with
         
        <code>
          setStyleUrl();
        </code>
        :
      </p>
      <pre
        className=" language-java"
      >
        <code
          className=" language-java"
        >
          mapboxMap
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            setStyle
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token keyword"
          >
            new
          </span>
           
          <span
            className="token class-name"
          >
            Style
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token class-name"
          >
            Builder
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token punctuation"
          >
            )
          </span>
          <span
            className="token punctuation"
          >
            .
          </span>
          <span
            className="token function"
          >
            fromUri
          </span>
          <span
            className="token punctuation"
          >
            (
          </span>
          <span
            className="token string"
          >
            "mapbox://styles/your-mapbox-username/your-style-ID"
          </span>
        </code>
      </pre>
    </div>
  </div>
</div>
`;
